pkg/api/helpers/download.go

```diff
@@ -55,6 +62,50 @@ func NewDownloader(client *apigen.ClientWithResponses, preSign bool) *Downloader
 	}
 }
 
+// DownloadWithObjectInfo downloads an object from lakeFS using pre-fetched object information,
+// avoiding the need for a separate stat call.
+func (d *Downloader) DownloadWithObjectInfo(ctx context.Context, src uri.URI, dst string, tracker *progress.Tracker, objectStat *apigen.ObjectStats) error {
+	// delete destination file if it exists
+	if err := os.Remove(dst); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove existing destination file '%s': %w", dst, err)
+	}
+
+	// create destination dir if needed
+	dir := filepath.Dir(dst)
+	_ = os.MkdirAll(dir, os.ModePerm)
+
+	// If symlink support is enabled, check if the object is a symlink and create it if so
+	if d.SymlinkSupport {
+		symlinkTarget, found := objectStat.Metadata.Get(apiutil.SymlinkMetadataKey)
+		if found && symlinkTarget != "" {
+			if tracker != nil {
+				tracker.UpdateTotal(0)
+				tracker.MarkAsDone()
+			}
+			// Skip non-regular files
+			if d.SkipNonRegularFiles {
+				return nil
+			}
+			// Create symlink instead of downloading file content
+			return os.Symlink(symlinkTarget, dst)
+		}
+		// fallthrough to download the object
+	}
+
+	// download object
+	var err error
+	if d.PreSign && swag.Int64Value(objectStat.SizeBytes) >= d.PartSize {
+		// download using presigned multipart download, it will fall back to presign single object download if needed
```

Comment:
When will it fallback to presign single object download?

Reply:
Cloud return an error in case the object `size < d.PartSize`, but instead it will just call `downloadObject`.
I've added a check here, before the call just not to go through the fallback.
Both are internal methods - I cloud just return an error in downloadPresignMultipart, wdyt?

Reply:
If `downloadPresignMultipart` is only called from here, you can remove the internal check in `downloadPresignMultipart`. If it's not, then leave it as is